src/screens/homeScreen.tsx

```diff
@@ -1,27 +1,114 @@
 // Import libraries and components
 import React, { Component, createRef } from 'react';
-import { View, Dimensions, StyleSheet } from 'react-native';
-import { DataTable, Card, TextInput, FAB } from 'react-native-paper';
+import { View, Dimensions, StyleSheet, Text } from 'react-native';
+import { DataTable, Card, TextInput, FAB, ActivityIndicator } from 'react-native-paper';
 import { StackNavigationProp } from '@react-navigation/stack';
 import { FlatList } from 'react-native-gesture-handler';
 import Fuse from 'fuse.js'
 
 // Import custom component
 import Map from '../components/map';
+import { TableData } from '../components/tableData';
+import { LoadingScreen } from '../components/loadingScreen';
 
-//  Import banner data from local file
-import bannerData from '../data/banners.json';
+// Import external values
+import SETTINGS from '../config/settings';
+import SECRETS from '../config/secrets';
 
 // Import typescript values
 import { ScreenStackParams } from '../typescript/types/screenparams';
 import { Screens } from '../typescript/enumerations/screens';
 import Banner from '../typescript/interfaces/banner';
-import { TableData } from '../components/tableData';
 
+// Functions to handle filtering
+/**
+   * Filters out banners that are disabled from array of banners
+   * @param banners Array of banners to be filtered
+   * @returns filtered list of banners
+   */
+const filterDisabledBanners = (banners: Banner[]): Banner[] => {
+  return [...banners].filter((banner) => {
```

Comment:
pretty sure you can one liner this:

return [...banners].filter((banner) => banner.enabled === true);